packages/devtools-proxy-support/src/system-ca.ts

```diff
@@ -150,14 +216,14 @@ export async function systemCA(
       !!process.env.DEVTOOLS_ALLOW_CERTIFICATES_WITHOUT_ISSUER
     )
   ) {
-    const reducedList = removeCertificatesWithoutIssuer(systemCerts);
-    systemCerts = reducedList.ca;
-    messages = messages.concat(reducedList.messages);
+    systemCerts = removeCertificatesWithoutIssuer(systemCerts, messages);
   }
 
   return {
     ca: mergeCA(
-      systemCerts,
+      sortByExpirationDate(systemCerts).map((cert) => {
```

Comment:
One thing I considered but haven't pursued is sorting all the certs we're passing here, I don't think it will have a major effect, but it seems like it would be closer to what's happening when openssl gets a list of certs passed. I had to restructure the code and split the functions in a way that should allow to do it pretty easy if we decide to change that.

Also not exactly sure about the ordering of passed arguments here, taking into account that it seems like it's important in what order certs are being passed to TLS connection, it might be that we should reverse the order and give priority to whatever user is passing directly through connection form, then certs bundled with Node.js, and only then the system certs